test/websocket/test_websocket_send.c

```diff
@@ -15,43 +21,50 @@ EM_BOOL WebSocketOpen(int eventType, const EmscriptenWebSocketOpenEvent *e, void
 }
 
 EM_BOOL WebSocketClose(int eventType, const EmscriptenWebSocketCloseEvent *e, void *userData) {
-  printf("close(eventType=%d, wasClean=%d, code=%d, reason=%s, userData=%p)\n", eventType, e->wasClean, e->code, e->reason, userData);
+  printf("close(socket=%d, eventType=%d, wasClean=%d, code=%d, reason=%s, userData=%p)\n", e->socket, eventType, e->wasClean, e->code, e->reason, userData);
   return 0;
 }
 
 EM_BOOL WebSocketError(int eventType, const EmscriptenWebSocketErrorEvent *e, void *userData) {
-  printf("error(eventType=%d, userData=%p)\n", eventType, userData);
+  printf("error(socket=%d, eventType=%d, userData=%p)\n", e->socket, eventType, userData);
   return 0;
 }
 
 EM_BOOL WebSocketMessage(int eventType, const EmscriptenWebSocketMessageEvent *e, void *userData) {
-  printf("message(eventType=%d, userData=%p data=%p, numBytes=%d, isText=%d)\n", eventType, userData, e->data, e->numBytes, e->isText);
+  printf("message(socket=%d, eventType=%d, userData=%p data=%p, numBytes=%d, isText=%d)\n", e->socket, eventType, userData, e->data, e->numBytes, e->isText);
   static int text_received = 0;
+  static int binary_received = 0;
+  assert(e->socket == sock1 || e->socket == sock2);
   if (e->isText) {
     printf("text data: \"%s\"\n", e->data);
     assert(strcmp((const char*)e->data, "hello on the other side") == 0);
-    text_received = 1;
-    return 0;
-  }
-
-  // We expect to receive the text message beofre the binary one
-  assert(text_received);
```

Comment:
With this assertion removed, this variable is no longer needed AFAICT? Nor is `binary_received`.